src/python/pants/engine/fs.py

```diff
@@ -73,6 +73,14 @@ class FileEntry:
     is_executable: bool = False
 
 
+@dataclass(frozen=True)
+class SymlinkEntry:
+    """A symlink pointing to a target path."""
```

Comment:
A bit more explanation of the constraints of this type (possibly by referring to the REAPI spec) might be good. Does the target need to be absolute, normalized, etc?